src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -53,50 +56,81 @@ public Optional<String> getActiveCertificateId(String certificatePem) {
                         clientFactory.getGreengrassV2DataClient().verifyClientDeviceIdentity(request);
                 return Optional.of(response.clientDeviceCertificateId());
             } catch (ValidationException | ResourceNotFoundException e) {
-                logger.atWarn().cause(e).kv("certificatePem", certificatePem)
+                logger.atWarn().cause(e).kv(CERTPEM_KEY, certificatePem)
                         .log("Certificate doesn't exist or isn't active");
                 return Optional.empty();
             } catch (Exception e) {
-                logger.atError().cause(e).kv("certificatePem", certificatePem)
+                logger.atError().cause(e).kv(CERTPEM_KEY, certificatePem)
                         .log("Failed to verify client device identity with cloud. Check that the core device's IoT "
                                 + "policy grants the greengrass:VerifyClientDeviceIdentity permission.");
                 throw new CloudServiceInteractionException("Failed to verify client device identity", e);
             }
         }
 
+        @Override
+        @SuppressWarnings("PMD.AvoidCatchingGenericException")
+        public Certificate getIotCertificate(String certificatePem) throws InvalidCertificateException {
+            if (Utils.isEmpty(certificatePem)) {
+                throw new IllegalArgumentException("Certificate PEM is empty");
+            }
+
+            // Throws InvalidCertificateException if we can't parse the certificate
+            Certificate cert = Certificate.fromPem(certificatePem);
+
+            VerifyClientDeviceIdentityRequest request =
+                    VerifyClientDeviceIdentityRequest.builder().clientDeviceCertificate(certificatePem).build();
+            try {
+                // We can ignore the response since it contains only the cert ID, which we directly compute
+                clientFactory.getGreengrassV2DataClient().verifyClientDeviceIdentity(request);
+                cert.setStatus(Certificate.Status.ACTIVE);
+            } catch (ValidationException | ResourceNotFoundException e) {
+                logger.atWarn().cause(e).kv(CERTPEM_KEY, certificatePem)
+                        .log("Certificate doesn't exist or isn't active");
+                throw new InvalidCertificateException("Invalid IoT certificate", e);
```

Comment:
Instead of throwing `InvalidCertificateException`, update cert status to `INACTIVE` here?

Reply:
That implies the certificate exists in the customer account, which is not the case.

Reply:
Mmmm I think there is a distinction between a certificate not existing or not being active. Following @vaibhavmurkute comment if it is not active it might be better to just set it as inactive and throw an error if it is not found. 

I know also this might be a debated suggestion but if would be nice to model this in terms of states. I see in your commented code that we catch it so it is something that is OK to happen. Why not return a result instead?.  So

* If it is empty return just `Result.ok()` you could add a method on the Result called `isEmpty` if you'd like (that represent an ok state but nothing returned.
* For the scenario of an invalid certificate maybe  having `cert.setStatus(Invalid)` `Result.ok(cert)` better represents the state than an exception
* Finally if it is valid we `cert.setStatus(Active)` return `Result.ok(cert)`

Reply:
I'm open to making tweaks to this in the future, but for now I don't want to model this if we don't get a positive response. The cloud API doesn't give us enough information to do this.